components/Elevation/src/UIView+MaterialElevationResponding.m

```diff
@@ -0,0 +1,70 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "UIView+MaterialElevationResponding.h"
+
+#import "MDCElevation.h"
+
+@implementation UIView (MaterialElevationResponding)
+
+- (CGFloat)mdc_baseElevation {
+  CGFloat totalElevation = 0;
+
+  if ([self conformsToProtocol:@protocol(MDCElevation)]) {
+    id<MDCElevation> elevatableSelf = (id<MDCElevation>)self;
+    if ([elevatableSelf respondsToSelector:@selector(mdc_overrideBaseElevation)]) {
+      return elevatableSelf.mdc_overrideBaseElevation;
+    }
+  }
+
+  if (!self.superview) {
```

Comment:
this conditional can be removed as the `while` takes care of this

Reply:
Updated this.